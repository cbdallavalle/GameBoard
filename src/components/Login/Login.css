#submit-btn {
  min-width: 311px;
  width: 88%;
  max-width: 350px;
  height: 42px;
  margin: 15px 0 7px 0;
  border: 2px solid #9b9a9a;
  background-color: #f4f4f4a6;
  font-size: 17px;
  font-family: 'Open Sans', sans-serif;
}

h1 {
  font-family: 'Righteous', cursive;
  font-size: 81px;
  margin-bottom: 15px;
  color: #313131;
}

form {
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.create-account {
  width: 100%;
}

.create-account input {
  margin: 15px 0 0 0;
}

.login-input {
  padding-left: 15px;
  min-width: 290px;
  width: 83%;
  max-width: 332px;;
  height: 35px;
  margin-top: 15px;
}

.login-input::placeholder {
  font-size: 13px;
}

#toggle-account {
  background-color: #ffffff7a;
  width: 245px;
  padding: 8px;
  border-radius: 11px;
  margin: 10px auto 15px auto;
}

#dice-img {
  width: 200px;
}

#pointer {
  margin-left: 4px;
}

#toggle-account:hover {
  cursor: pointer;
}

@media screen and (min-width: 450px) {
  h1 {
    font-size: 90px;
  }

  #dice-img {
    width: 300px;
  }
}