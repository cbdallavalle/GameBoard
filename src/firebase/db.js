import { db } from './firebase';

export const doCreateUser = (id, firstName, lastName, email) =>
  db.ref(`users/${id}`).set({
    user: {
      id,
      firstName,
      lastName,
      email,
    }
  });

export const onceGetUsers = async () => {
  try {
    const snapshot = await db.ref('users').once('value');
    if (snapshot.status < 300) {
      const value = await snapshot.val();
      return value
    } else {
      throw new Error('unable to fetch users')
    }
  } catch (error) {
    throw new Error('unable to fetch users')
  }
}

export const doWriteFavoriteData = async (userId, newFavorites) => {
  try {
    db.ref('users/' + userId + '/favorites').set(newFavorites);
  } catch (error) {
    throw new Error('unable to write favorite data')
  }
}

export const doAddFavoriteData = async (userId, favorite) => {
  try {
    const existingFavorites = await getFavorites(userId);
    const favoriteToAdd = checkFavoriteDuplication(existingFavorites, favorite);
    doWriteLastFavoritedData(userId, favorite)
    const newFavorites = {...existingFavorites, ...favoriteToAdd }
    doWriteFavoriteData(userId, newFavorites)
  } catch (error) {
    throw new Error('unable to add favorite data')
  }
}

export const doDeleteFavoriteData = async (userId, favorite) => {
  try {
    const existingFavorites = await getFavorites(userId);
    delete existingFavorites[favorite.name]
    doWriteFavoriteData(userId, existingFavorites)
  } catch (error) {
    throw new Error('unable to delete')
  }
}

export const doWriteLastFavoritedData = async (userId, favorite) => {
  db.ref('users/' + userId + '/lastFavorited').set(favorite);
}

export const checkFavoriteDuplication = (existingFavorites, favorite) => {
  const { name, id, description, image, thumbnail } = favorite;
  const review = "No review - click to add one";
  const favoriteToAdd = { [name]: { id, description, image, thumbnail, name, review } }
  return existingFavorites && existingFavorites[favorite.name] ? {} : favoriteToAdd
}

export const doWriteFriendsData = async (userId, friendId) => {
  try {
    const friends = await getFriends(userId);
    if (friends && !friends.includes(friendId)) {
      friends.push(friendId);
      db.ref('users/' + userId + '/friends').set(friends); 
    } else if (!friends) {
      db.ref('users/' + userId + '/friends').set([friendId]); 
    }
  } catch (error) {
    throw new Error ('unable to add friend')
  }
}

export const getFriends = async (userId) => {
  try {
    const value = await onceGetUsers();
    return value[userId].friends
  } catch (error) {
    throw new Error('unable to load friends data')
  }
}

export const getFavorites = async (userId) => {
  try {
    const value = await onceGetUsers();
    return value[userId].favorites;
  } catch (error) {
    throw new Error ('unable to load favorites data')
  }
}

export const getFriendsFavorites = async (userId) => {
  const friends = await getFriends(userId);
  const value = await onceGetUsers();
  if(friends) {
    return friends.reduce( (accu, friendId) => {
      if(value[friendId].lastFavorited) {
        accu[value[friendId].user.firstName] = value[friendId].lastFavorited
      }
      return accu
    }, {} )  
  } else {
    return {}
  }
}

export const doWriteReviewData = async (userId, favorite, review) => {
  try {
    const { name } = favorite;
    const reviewedGame = {...favorite, review}
    const response = await db.ref('users/' + userId + '/favorites/' + name).set(reviewedGame);
    if(response.status < 300) {
      await doWriteLastFavoritedData(userId, reviewedGame)
    } else {
      throw new Error('unable to process review')
    }
  } catch (error) {
    throw new Error('unable to process review')
  }
}