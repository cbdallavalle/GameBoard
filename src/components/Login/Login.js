import React, { Component } from 'react';
import dice from '../../assets/dice.gif';
import { auth, db } from '../../firebase';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { SignIn } from '../SignIn/SignIn';
import { SignUp } from '../SignUp/SignUp';
import './Login.css'

const initialState = {
  name: '',
  email: '',
  passwordOne: '',
  passwordTwo: '',
  error: null,
  displayCreate: 'login',
}

export class Login extends Component {
  constructor() {
    super();
    this.state = {  ...initialState }
  }

  handleChange = e => {
    e.preventDefault();
    this.setState({ [e.target.name]: e.target.value })
  }

  determineDisplay = () => {
    return this.state.displayCreate === 'signup' ?
    <div>
      <SignUp 
        handleChange={ this.handleChange }
        name={ this.state.name }
        email={ this.state.email }
        passwordOne={ this.state.passwordOne }
        passwordTwo={ this.state.passwordTwo }
      />
      <p 
        id="toggle-account"
        onClick={ this.toggleDisplay }
      >
        Already have an account? Log in!      
        <i className="fas fa-mouse-pointer" id="pointer"></i>
      </p>
    </div>
    :
    <div>
      <SignIn 
        handleChange={ this.handleChange }
        email={ this.state.email }
        passwordOne={ this.state.passwordOne }
      />
      <p 
        id="toggle-account"
        onClick={ this.toggleDisplay }
      >
        Don't have an account? Create one! 
        <i className="fas fa-mouse-pointer" id="pointer"></i>
      </p>
    </div>
  }

  toggleDisplay = () => {
    const displayCreate = this.state.displayCreate === 'login' ? 'signup' : 'login';
    this.setState({ displayCreate, name: '' })
  }

  handleSubmit = event => {
    event.preventDefault();
    const {
      email,
      passwordOne,
    } = this.state;
    this.state.name ? this.signUp(email, passwordOne) : this.logIn(email, passwordOne);
  }

  signUp = async (email, passwordOne) => {
    const { history } = this.props;

    try {
      const user = await auth.doCreateUserWithEmailAndPassword(email, passwordOne);
      await db.doCreateUser(user.uid, this.state.name, email)
      this.setState({ ...initialState })
      history.push('/');        
    } catch(error) {
      this.setState({ error: error.message })
    }
  }

  logIn = async (email, password) => {
    const { history } = this.props;

    try {
      await auth.doSignInWithEmailAndPassword(email, password);
      this.setState({ ...initialState })
      history.push('/');
    } catch(error) {
      this.setState({ error: error.message })
    }
  }

  determineInvalid = () => {
    if(this.state.displayCreate === 'login') {
      return (this.state.email === '' || this.state.passwordOne === '');
    } else {
      return (
        this.state.passwordOne !== this.state.passwordTwo ||
        this.state.passwordOne === '' ||
        this.state.email === '' ||
        this.state.name === ''
      )
    }
  }

  render() {
    return (
      <form action="submit" onSubmit={this.handleSubmit}>
        <img src={ dice } alt="dice-logo" id="dice-img"/>
        <h1>Game Board</h1>
        { this.determineDisplay() }
        <button 
          disabled={this.determineInvalid()} 
          className="LogIn" 
          id="submit-btn"
        >
          Log In!
        </button>
        { this.state.error && <p>{this.state.error}</p> }
      </form>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user
})

export default withRouter(connect(mapStateToProps, null)(Login));