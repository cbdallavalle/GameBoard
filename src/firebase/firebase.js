import * as firebase from 'firebase';

const config = {
  apiKey: "AIzaSyBDU0_Pa8xdAWKmtKKfo3QkbrtR56cl024",
  authDomain: "dallavalle-gameboard.firebaseapp.com",
  databaseURL: "https://dallavalle-gameboard.firebaseio.com",
  projectId: "dallavalle-gameboard",
  storageBucket: "",
  messagingSenderId: "250758219123"
};

if(!firebase.apps.length) {
  firebase.initializeApp(config)
}

const auth = firebase.auth();

export {
  auth,
}