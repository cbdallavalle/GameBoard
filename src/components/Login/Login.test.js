import React from 'react';
import { Login } from './Login';
import { shallow } from 'enzyme';
import { auth, db } from '../../firebase';

describe("Login", () => {
  const initialState = {
    firstName: '',
    lastName: '',
    email: '',
    passwordOne: '',
    passwordTwo: '',
    error: null,
    displayCreate: 'login',
  }

  const erroredSignInState = {
    "displayCreate": "login", 
    "email": "", 
    "error": "signInWithEmailAndPassword failed: First argument \"email\" must be a valid string.", 
    "firstName": "", 
    "lastName": "",
    "passwordOne": "", 
    "passwordTwo": ""
  }

  const erroredSignUpState = {
    "displayCreate": "login", 
    "email": "", 
    "error": "createUserWithEmailAndPassword failed: First argument \"email\" must be a valid string.", 
    "firstName": "", 
    "lastName": "",
    "passwordOne": "", 
    "passwordTwo": ""
  }

  it('should have a default state', () => {
    const wrapper = shallow(<Login user={{}} />)
    expect(wrapper.state()).toEqual(initialState)
  })

  it('should match the snapshot when displayCreate is login', () => {
    const wrapper = shallow(<Login user={{}} />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when displayCreate is signup', () => {
    const wrapper = shallow(<Login user={{}} />)
    wrapper.setState({ displayCreate: 'signup' })
    expect(wrapper).toMatchSnapshot();
  })

  it('should toggle the display to signup', () => {
    const wrapper = shallow(<Login user={{}} />)
    expect(wrapper.state().displayCreate).toEqual('login');
    wrapper.instance().toggleDisplay();
    expect(wrapper.state().displayCreate).toEqual('signup');
  })

  it('should toggle the display to login', () => {
    const wrapper = shallow(<Login user={{}} />)
    wrapper.setState({ displayCreate: 'signup' })
    expect(wrapper.state().displayCreate).toEqual('signup');
    wrapper.instance().toggleDisplay();
    expect(wrapper.state().displayCreate).toEqual('login');
  })

  it('should call signUp when state has name', async() => {
    const wrapper = shallow(<Login user={{}} />)
    const mockEvent = {preventDefault: () => {}}
    wrapper.instance().signUp = jest.fn();
    await wrapper.setState({ firstName: 'first', email: 'email', passwordOne: 'passwordOne' })
    wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.instance().signUp).toHaveBeenCalledWith('email', 'passwordOne');
  })

  it('should call logIn when state does not have name', async() => {
    const wrapper = shallow(<Login user={{}} />)
    const mockEvent = {preventDefault: () => {}}
    wrapper.instance().logIn = jest.fn();
    await wrapper.setState({ email: 'email', passwordOne: 'passwordOne' })
    wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.instance().logIn).toHaveBeenCalledWith('email', 'passwordOne');
  })

  it('when signUp called it should set state with error message when call fails', () => {
    const wrapper = shallow(<Login user={{}} />);
    wrapper.instance().signUp();
    expect(wrapper.state()).toEqual(erroredSignUpState)
  })

  it('when signUp called it should set state with initial state when succesful', () => {
    auth.doCreateUserWithEmailAndPassword = jest.fn();
    db.doCreateUser = jest.fn();
    const wrapper = shallow(<Login user={{}} />)
    wrapper.instance().signUp({email: 'email'}, {passwordOne: 'passwordOne'});
    expect(wrapper.state()).toEqual(initialState)
  })

  it('when logIn called it should set state with error message when call fails', () => {
    const wrapper = shallow(<Login user={{}} />)
    wrapper.instance().logIn();
    expect(wrapper.state()).toEqual(erroredSignInState)
  })

  it('when logIn should should set state with initial state when successful', async() => {
    auth.doSignInWithEmailAndPassword = jest.fn();
    const wrapper = shallow(<Login user={{}} />)
    wrapper.instance().logIn({email: 'email'}, {passwordOne: 'passwordOne'});
    expect(wrapper.state()).toEqual(initialState)
  })

  it('determineInvalid should return false if state is login and requirements are met', () => {
    const wrapper = shallow(<Login user={{}} />)
    wrapper.setState({ passwordOne: 'pass', email: 'email' })
    expect(wrapper.instance().determineInvalid()).toEqual(false)
  })

  it('determineInvalid should return true if state is login and requirements are met', () => {
    const wrapper = shallow(<Login user={{}} />)
    expect(wrapper.instance().determineInvalid()).toEqual(true)
  })

  it('determineInvalid should return false if state is signup and requirements are met', () => {
    const wrapper = shallow(<Login user={{}} />)
    wrapper.setState({ displayCreate: 'signup', passwordOne: 'pass', passwordTwo: 'pass', email: 'email', firstName: 'name', lastName: 'name' })
    expect(wrapper.instance().determineInvalid()).toEqual(false)

  })

  it('determineInvalid should return true if state is signup and requirements are met', () => {
    const wrapper = shallow(<Login user={{}} />)
    wrapper.setState({ displayCreate: 'signup' })
    expect(wrapper.instance().determineInvalid()).toEqual(true)
  })

})