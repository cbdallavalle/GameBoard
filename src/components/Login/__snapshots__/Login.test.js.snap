// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login determineDisplay should match the snapshot when displayCreate is login 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "action": "submit",
      "children": Array [
        <img
          alt="dice-logo"
          id="dice-img"
          src="dice.gif"
/>,
        <h1>
          Game Board
</h1>,
        <div>
          <Unknown
                    email=""
                    handleChange={[Function]}
                    passwordOne=""
          />
          <p
                    id="toggle-account"
                    onClick={[Function]}
          >
                    Don't have an account? Create one!
                    <i
                              className="fas fa-mouse-pointer"
                              id="pointer"
                    />
          </p>
</div>,
        <button
          className="LogIn"
          disabled={true}
          id="submit-btn"
          type="submit"
>
          Submit
</button>,
        null,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "dice-logo",
          "id": "dice-img",
          "src": "dice.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Game Board",
        },
        "ref": null,
        "rendered": "Game Board",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              email=""
              handleChange={[Function]}
              passwordOne=""
/>,
            <p
              id="toggle-account"
              onClick={[Function]}
>
              Don't have an account? Create one!
              <i
                            className="fas fa-mouse-pointer"
                            id="pointer"
              />
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "email": "",
              "handleChange": [Function],
              "passwordOne": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Don't have an account? Create one!",
                <i
                  className="fas fa-mouse-pointer"
                  id="pointer"
/>,
              ],
              "id": "toggle-account",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Don't have an account? Create one!",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-mouse-pointer",
                  "id": "pointer",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "LogIn",
          "disabled": true,
          "id": "submit-btn",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
      null,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "submit",
        "children": Array [
          <img
            alt="dice-logo"
            id="dice-img"
            src="dice.gif"
/>,
          <h1>
            Game Board
</h1>,
          <div>
            <Unknown
                        email=""
                        handleChange={[Function]}
                        passwordOne=""
            />
            <p
                        id="toggle-account"
                        onClick={[Function]}
            >
                        Don't have an account? Create one!
                        <i
                                    className="fas fa-mouse-pointer"
                                    id="pointer"
                        />
            </p>
</div>,
          <button
            className="LogIn"
            disabled={true}
            id="submit-btn"
            type="submit"
>
            Submit
</button>,
          null,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "dice-logo",
            "id": "dice-img",
            "src": "dice.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Game Board",
          },
          "ref": null,
          "rendered": "Game Board",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                email=""
                handleChange={[Function]}
                passwordOne=""
/>,
              <p
                id="toggle-account"
                onClick={[Function]}
>
                Don't have an account? Create one!
                <i
                                className="fas fa-mouse-pointer"
                                id="pointer"
                />
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "email": "",
                "handleChange": [Function],
                "passwordOne": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Don't have an account? Create one!",
                  <i
                    className="fas fa-mouse-pointer"
                    id="pointer"
/>,
                ],
                "id": "toggle-account",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Don't have an account? Create one!",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-mouse-pointer",
                    "id": "pointer",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "LogIn",
            "disabled": true,
            "id": "submit-btn",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Login determineDisplay should match the snapshot when displayCreate is signup 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "action": "submit",
      "children": Array [
        <img
          alt="dice-logo"
          id="dice-img"
          src="dice.gif"
/>,
        <h1>
          Game Board
</h1>,
        <div>
          <Unknown
                    email=""
                    firstName=""
                    handleChange={[Function]}
                    lastName=""
                    passwordOne=""
                    passwordTwo=""
          />
          <p
                    id="toggle-account"
                    onClick={[Function]}
          >
                    Already have an account? Log in!
                    <i
                              className="fas fa-mouse-pointer"
                              id="pointer"
                    />
          </p>
</div>,
        <button
          className="LogIn"
          disabled={true}
          id="submit-btn"
          type="submit"
>
          Submit
</button>,
        null,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "dice-logo",
          "id": "dice-img",
          "src": "dice.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Game Board",
        },
        "ref": null,
        "rendered": "Game Board",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              email=""
              firstName=""
              handleChange={[Function]}
              lastName=""
              passwordOne=""
              passwordTwo=""
/>,
            <p
              id="toggle-account"
              onClick={[Function]}
>
              Already have an account? Log in!
              <i
                            className="fas fa-mouse-pointer"
                            id="pointer"
              />
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "email": "",
              "firstName": "",
              "handleChange": [Function],
              "lastName": "",
              "passwordOne": "",
              "passwordTwo": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Already have an account? Log in!",
                <i
                  className="fas fa-mouse-pointer"
                  id="pointer"
/>,
              ],
              "id": "toggle-account",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Already have an account? Log in!",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-mouse-pointer",
                  "id": "pointer",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "LogIn",
          "disabled": true,
          "id": "submit-btn",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
      null,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "submit",
        "children": Array [
          <img
            alt="dice-logo"
            id="dice-img"
            src="dice.gif"
/>,
          <h1>
            Game Board
</h1>,
          <div>
            <Unknown
                        email=""
                        firstName=""
                        handleChange={[Function]}
                        lastName=""
                        passwordOne=""
                        passwordTwo=""
            />
            <p
                        id="toggle-account"
                        onClick={[Function]}
            >
                        Already have an account? Log in!
                        <i
                                    className="fas fa-mouse-pointer"
                                    id="pointer"
                        />
            </p>
</div>,
          <button
            className="LogIn"
            disabled={true}
            id="submit-btn"
            type="submit"
>
            Submit
</button>,
          null,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "dice-logo",
            "id": "dice-img",
            "src": "dice.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Game Board",
          },
          "ref": null,
          "rendered": "Game Board",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                email=""
                firstName=""
                handleChange={[Function]}
                lastName=""
                passwordOne=""
                passwordTwo=""
/>,
              <p
                id="toggle-account"
                onClick={[Function]}
>
                Already have an account? Log in!
                <i
                                className="fas fa-mouse-pointer"
                                id="pointer"
                />
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "email": "",
                "firstName": "",
                "handleChange": [Function],
                "lastName": "",
                "passwordOne": "",
                "passwordTwo": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Already have an account? Log in!",
                  <i
                    className="fas fa-mouse-pointer"
                    id="pointer"
/>,
                ],
                "id": "toggle-account",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Already have an account? Log in!",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-mouse-pointer",
                    "id": "pointer",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "LogIn",
            "disabled": true,
            "id": "submit-btn",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
